runtime/src/main/java/org/corfudb/runtime/view/ObjectsView.java

```diff
@@ -157,6 +157,17 @@ public long TXEnd() throws TransactionAbortedException {
             long commitAddress = TransactionalContext.getCurrentContext().commitTransaction();
             MicroMeterUtils.time(Duration.ofMillis(System.currentTimeMillis() - context.getStartTime()),
                     "transaction.duration");
+            if (commitAddress == Address.NON_ADDRESS) {
+                // If no streams were touched or only empty streams were touched, the returned address would be -1
+                // But -1 can be detrimental to stream subscription which is just looking for a safe spot
+                // to resume subscription from, so instead return the address from the snapshot token taken.
+                long snapshotAddress = context.getSnapshotTimestamp().getSequence();
+                if (snapshotAddress > 0) {
```

Comment:
This is incorrect because the snapshot can regress from 0 to less than 0, which misses the first update. The condition should be `if (snapshotAddress >= 0)` and NOT  `if (snapshotAddress > 0)`

Reply:
Done. thanks